tests/lib/rules/function-component-definition.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @fileoverview Standardize the way function component get defined
+ * @author Stefan Wullems
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/function-component-definition');
+
+const parserOptions = {
+  ecmaVersion: 2018,
+  sourceType: 'module',
+  ecmaFeatures: {
+    jsx: true
+  }
+};
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({parserOptions});
+ruleTester.run('function-component-definition', rule, {
+
+  valid: [{
+    code: `
+        class Hello extends React.Component {
+          render () {
+            return <div>Hello {this.props.name}</div>;
+          }
+        }
+      `,
+    options: ['arrow-function']
+  }, {
+    code: `
+        class Hello extends React.Component {
+          render () {
+            return <div>Hello {this.props.name}</div>;
+          }
+        }
+      `,
+    options: ['function-declaration']
+  }, {
+    code: `
+        class Hello extends React.Component {
+          render () {
+            return <div>Hello {this.props.name}</div>;
+          }
+        }
+      `,
+    options: ['function-expression']
+  }, {
+    code: `
+      const Hello = (props) => {
+        return <div/>;
+      }
+    `,
+    options: ['arrow-function']
+  }, {
+    code: `
+      function Hello (props) {
+        return <div/>;
+      }
+    `,
+    options: ['function-declaration']
+  }, {
+    code: `
+      const Hello = function (props) {
+        return <div/>;
+      }
+    `,
+    options: ['function-expression']
+  }],
```

Comment:
need test cases for valid `const Foo = React.memo(function Foo() { returns <p />})` when option is `function-declaration`.